### PR TITLE
Normalise Directive & Filter functionality + support `mode` parameter

### PR DESCRIPTION
**Pull request by tthew**
Addresses #1 
- Refactor directive & filter to normalise functionality
- Adds 'mode' parameter with option of 'first' (default) and 'words'
- Improves test coverage
- Improves documentation

### CODE REVIEW

File: tests/capitalize.filter.spec.js [hunk 1]
```diff
@@ -1,16 +1,74 @@
...
+      scope = _$rootScope_;
+    }));
+
+    it('should capitalize first letter by default', function() {
+      el = '<p>{{ "i am a lower case string" | capitalize }}</p>';
+      $el = compile(el)(scope);
+      scope.$digest();
+      expect($el.text()).to.equal('I am a lower case string');
+    });
+
```
Comment (codebeige) [inline tests/capitalize.filter.spec.js:55]: In the future we should maybe consider further improvement for edge cases like the following:
    "FOO BAR"  => "Foo Bar"

**Comment by codebeige** [comment]
:+1: Great! That was fast. Looking forward to using it inside `angular-symptoms` on Monday.

File: tests/capitalize.filter.spec.js (hunk 1 above)
Comment (tthew) [inline tests/capitalize.filter.spec.js:55]: Agreed.

**Comment by tthew** [comment]
Cool, didn't expect you to look at this until next week.  I'll merge and then make a release so it's ready for you.

**Comment by codebeige** [comment]
Did already do the merge, just wanted Travis to complete first. (You never know, do you? _g_)
